core/src/main/java/org/apache/iceberg/DataTableScan.java

```diff
@@ -71,7 +71,8 @@ protected TableScan newRefinedScan(TableOperations ops, Table table, Schema sche
   public CloseableIterable<FileScanTask> planFiles(TableOperations ops, Snapshot snapshot,
                                                    Expression rowFilter, boolean ignoreResiduals,
                                                    boolean caseSensitive, boolean colStats) {
-    ManifestGroup manifestGroup = new ManifestGroup(ops.io(), snapshot.dataManifests(), snapshot.deleteManifests())
+    ManifestGroup manifestGroup = new ManifestGroup(ops.io(), snapshot.dataManifests(), snapshot.deleteManifests(),
+        ops.current().location(), ops.current().properties())
```

Comment:
Why do we push the whole `properties`? Shall we just push a flag `useRelative`, or we can just push the `location` and if it is not null then we use it as a base for the relative paths?

Reply:
Instead of  pushing `properties` around, I introduced a flag `shouldUseRelativePaths`.